### PR TITLE
Feature law demo

### PR DESCRIPTION
**Pull request by wadeking98**
- added connectionless support to law proof request with BC Wallet
- added new character Joyce
- added accessibility to character selection screen
- added support to LSBC Proof requests from staging and mainNet

### CODE REVIEW

File: client/src/pages/onboarding/OnboardingContainer.tsx [hunk 1]
```diff
@@ -76,6 +76,75 @@ export const OnboardingContainer: React.FC<Props> = ({
     dispatch(prevOnboardingStep())
   }
 
+  //override title and text content to make them character dependant
+  const getCharacterContent = (progress: number) => {
+    const ret: { [key: number]: { [key: string]: { [key: string]: string } } } = {
+      [Progress.PICK_CHARACTER]: {
+        Alice: {
```
Comment (amanji) [inline client/src/pages/onboarding/OnboardingContainer.tsx:83]: Shouldn't this be configurable from the list of characters in the backend?

Comment (wadeking98) [reply client/src/pages/onboarding/OnboardingContainer.tsx]: You're right, I moved it to the backend now

File: client/src/pages/onboarding/OnboardingContainer.tsx [hunk 2]
```diff
@@ -76,6 +76,16 @@ export const OnboardingContainer: React.FC<Props> = ({
     dispatch(prevOnboardingStep())
   }
 
+  //override title and text content to make them character dependant
+  const getCharacterContent = (progress: number) => {
+    if (currentCharacter?.content[progress]) {
```
Comment (amanji) [inline client/src/pages/onboarding/OnboardingContainer.tsx:81]: Would it be cleaner to do something like:

```js
  const charcterContent = currentCharacter?.content[progress]
  if (characterContent) {
    return characterContent)
  }
  return { ... }

File: client/src/pages/onboarding/OnboardingContainer.tsx [hunk 3]
```diff
@@ -98,6 +108,8 @@ export const OnboardingContainer: React.FC<Props> = ({
           content={OnboardingContent[progress]}
           currentCharacter={currentCharacter}
           characters={characters}
+          title={getCharacterContent(progress).title}
```
Comment (amanji) [inline client/src/pages/onboarding/OnboardingContainer.tsx:111]: Could use a destructuring assignment like: `const {title, text} = getCharacterContent(progress)`

Then just pass `title` and `text` in the prop.

File: client/src/pages/onboarding/steps/AcceptCredential.tsx [hunk 4]
```diff
@@ -121,7 +130,7 @@ export const AcceptCredential: React.FC<Props> = ({ content, connectionId, crede
 
   return (
     <motion.div className="flex flex-col h-full" variants={fadeX} initial="hidden" animate="show" exit="exit">
-      <StepInformation title={content.title} text={content.text} />
+      <StepInformation title={title ? title : content.title} text={text ? text : content.text} />
```
Comment (amanji) [inline client/src/pages/onboarding/steps/AcceptCredential.tsx:133]: Simpler to just say `title ?? content.title`

**Review by amanji — CHANGES_REQUESTED** [review]

File: client/src/pages/onboarding/OnboardingContainer.tsx (hunk 2 above)
Comment (wadeking98) [reply client/src/pages/onboarding/OnboardingContainer.tsx]: change completed

File: client/src/pages/onboarding/steps/AcceptCredential.tsx (hunk 4 above)
Comment (wadeking98) [reply client/src/pages/onboarding/steps/AcceptCredential.tsx]: change completed

File: server/src/content/businessWoman/useCases/Hotel.ts [hunk 5]
```diff
@@ -4,36 +4,40 @@ import { v4 as uuid } from 'uuid'
```
Comment (amanji) [inline server/src/content/businessWoman/useCases/Hotel.ts:4]: Seems like this is identical to the student content

File: client/src/pages/onboarding/steps/SetupConnection.tsx [hunk 6]
```diff
@@ -84,7 +117,7 @@ export const SetupConnection: React.FC<Props> = ({ content, connectionId, invita
       animate="show"
       exit="exit"
     >
-      <StepInformation title={content.title} text={content.text} />
+      <StepInformation title={title ? title : content.title} text={text ? text : content.text} />
```
Comment (amanji) [inline client/src/pages/onboarding/steps/SetupConnection.tsx:120]: Could use nullish coalescing here too

Comment (wadeking98) [reply client/src/pages/onboarding/steps/SetupConnection.tsx]: change completed

**Review by amanji — APPROVED** [review]

**Pull request merged**